commands/publish_dafsa.py

```diff
@@ -0,0 +1,100 @@
+import os
+import json
+import tempfile
+import subprocess
+
+import requests
+from kinto_http import Client, KintoException
+
+
+PSL_FILENAME = "public_suffix_list.dat"
+
+COMMIT_HASH_URL = (
+    f"https://api.github.com/repos/publicsuffix/list/commits?path={PSL_FILENAME}"
+)
+LIST_URL = f"https://raw.githubusercontent.com/publicsuffix/list/master/{PSL_FILENAME}"
+
+MAKE_DAFSA_PY = "https://raw.githubusercontent.com/arpit73/temp_dafsa_testing_repo/master/publishing/make_dafsa.py"  # noqa
+PREPARE_TLDS_PY = "https://raw.githubusercontent.com/arpit73/temp_dafsa_testing_repo/master/publishing/prepare_tlds.py"  # noqa
+
+BUCKET_ID = os.getenv("BUCKET_ID", "main-workspace")
+COLLECTION_ID = "public-suffix-list"
+RECORD_ID = "latest-commit-hash"
+
+
+def get_latest_hash():
+    response = requests.get(COMMIT_HASH_URL)
+    response.raise_for_status()
+    return response.json()[0]["sha"]
+
+
+def download_resources(directory, *urls):
+    for url in urls:
+        file_name = os.path.basename(url)
+        file_location = os.path.join(directory, file_name)
+        response = requests.get(url, stream=True)
+        response.raise_for_status()
+
+        with open(file_location, "wb") as f:
+            for chunk in response.iter_content(chunk_size=1024):
+                f.write(chunk)
+
+
+def make_dafsa_and_publish(client, latest_hash):
+    with tempfile.TemporaryDirectory() as tmp:
+        download_resources(tmp, LIST_URL, MAKE_DAFSA_PY, PREPARE_TLDS_PY)
+        """
+        prepare_tlds.py is called with the two arguments the location of
+        the downloaded public suffix list and the name of the output file
+        """
+        output_binary_name = "etld_data.json"
+
+        output_binary_path = os.path.join(tmp, output_binary_name)
+        prepare_tlds_py_path = os.path.join(tmp, "prepare_tlds.py")
+        raw_psl_path = os.path.join(tmp, PSL_FILENAME)
+
+        # Make the DAFSA
+        run = subprocess.run(
+            ["python3", prepare_tlds_py_path, raw_psl_path, output_binary_path, "--bin"]
+        )
+        if run.returncode != 0:
+            raise Exception("DAFSA Build Failed !!!")
+
+        subprocess.run(["ls", tmp])
+
+        # Upload the attachment
+        mimetype = "application/octet-stream"
+        filecontent = open(output_binary_path, "rb").read()
+        record_uri = client.get_endpoint("record", id=RECORD_ID)
+        attachment_uri = f"{record_uri}/attachment"
+        multipart = [("attachment", (output_binary_name, filecontent, mimetype))]
+        commit_hash = json.dumps({"commit-hash": latest_hash})
+
+        client.session.request(
+            method="post", data=commit_hash, endpoint=attachment_uri, files=multipart
+        )
+        client.patch_collection(data={"status": "to-review"})
```

Comment:
This would deserve a comment.

Plus, the above `.request()` instruction should be grouped with the previous block instead of this one, don't you think?